### PR TITLE
CORE-18634 Add HSM category for session encryption

### PR DESCRIPTION
**Pull request by YashNabar**
Adds a new HSM category `ENCRYPTION_SECRET` which will be used for encrypting session data before storing the session in a shared storage. This change also updates the app-simulator scripts, end-to-end test utilities, and network CLI-plugin to assign the new category to the cluster-level tenant "p2p".

Plugin smoke tests: https://ci02.dev.r3.com/blue/organizations/jenkins/Corda5%2Fcorda-runtime-os-combined-worker-plugins-smoke-tests/detail/PR-5220/1/pipeline

Manual test:

Screenshot 2023-12-08 at 16 22 04

Using v5_1 path:

Screenshot 2023-12-08 at 16 20 53

### CODE REVIEW

**Review by yift-r3 — APPROVED** [review]
LGTM

**Review by kyriathar — DISMISSED** [review]
OK (crypto)

**Review by yift-r3 — DISMISSED** [review]

File: components/membership/membership-rest/src/main/kotlin/net/corda/membership/rest/v1/HsmRestResource.kt [hunk 1]
```diff
@@ -55,7 +55,7 @@ interface HsmRestResource : RestResource {
                 " tenant of the P2P services, or the value 'rest' for a cluster-level tenant of the REST")
         tenantId: String,
         @RestPathParameter(description = "The category of the HSM; can be the value 'ACCOUNTS', 'CI', 'LEDGER'," +
-                " 'NOTARY', 'SESSION_INIT', 'TLS', or 'JWT_KEY'")
+                " 'NOTARY', 'SESSION_INIT', 'TLS', 'JWT_KEY', or 'ENCRYPTION_SECRET'")
         category: String
     ): HsmAssociationInfo
```
Comment (vkolomeyko) [inline components/membership/membership-rest/src/main/kotlin/net/corda/membership/rest/v1/HsmRestResource.kt:103]: In order to preserve correct REST API versioning information from earlier versions of Corda, we need to leave this method unchanged and constrain it with `maxVersion = RestApiVersion.C5_1`.
In order to introduce new option, we should start a new method with `minVersion = RestApiVersion.C5_2`.

File: processors/rest-processor/src/integrationTest/resources/swaggerBaseline-v1.json [hunk 2]
```diff
@@ -1121,18 +1121,18 @@
         }, {
           "name" : "category",
           "in" : "path",
-          "description" : "The category of the HSM; can be the value 'ACCOUNTS', 'CI', 'LEDGER', 'NOTARY', 'SESSION_INIT', 'TLS', or 'JWT_KEY'",
```
Comment (vkolomeyko) [inline processors/rest-processor/src/integrationTest/resources/swaggerBaseline-v1.json]: This file should not change, please see my comment about REST API versioning.

File: processors/rest-processor/src/integrationTest/resources/swaggerBaseline-v5_1.json [hunk 3]
```diff
@@ -1121,18 +1121,18 @@
         }, {
           "name" : "category",
           "in" : "path",
-          "description" : "The category of the HSM; can be the value 'ACCOUNTS', 'CI', 'LEDGER', 'NOTARY', 'SESSION_INIT', 'TLS', or 'JWT_KEY'",
```
Comment (vkolomeyko) [inline processors/rest-processor/src/integrationTest/resources/swaggerBaseline-v5_1.json]: This file should not change, please see my comment about REST API versioning.

**Review by vkolomeyko — APPROVED** [review]
LGTM

**Review by yift-r3 — APPROVED** [review]

**Review by kyriathar — APPROVED** [review]

**Pull request merged**